include/matx/operators/permute.h

```diff
@@ -137,10 +146,27 @@ namespace matx
    * @return permuted operator
    */
   template <typename T>
-    __MATX_INLINE__ auto permute( const T op, 
+    __MATX_INLINE__ auto permute( const T &op, 
         const int32_t (&dims)[T::Rank()]) {
       return detail::PermuteOp<T>(op, dims);
     }
+
+  /**
+   * @brief Operator to permute the dimensions of a tensor or operator.
+   *
+   * The each dimension must appear in the dims array once.
+   * This operator can appear as an rvalue or lvalue. 
+   *
+   * @tparam T Input operator/tensor type
+   * @param op Input operator
+   * @param dims the reordered dimensions of the operator.
+   * @return permuted operator
+   */
+  template <typename T>
+    __MATX_INLINE__ auto permute( const T &op, 
+        const std::array<int32_t, T::Rank()> &dims) {
```

Comment:
instead of having two identical functions in the class the C array version should call detail::toarray() and just convert and can this function.

Reply:
fixed